File: 27_Number_Of_Candies/main.go
package main

import (
	"fmt"
)

func kidsWithCandies(candies []int, extraCandies int) []bool {
	var (
		result []bool
		max    = 0
	)

	for _, c := range candies {
		if c > max {
			max = c
		}
	}

	for _, c := range candies {
		if c+extraCandies >= max {
			result = append(result, true)
			continue
		}

		result = append(result, false)
	}

	return result
}

func main() {

	x := kidsWithCandies([]int{12, 1, 12}, 10)
	fmt.Println(x)
}

File: 27_Number_Of_Candies/main.js
function kidsWithCandies(candies, extraCandies) {
  let result = []
  let max = Math.max(...candies)

  for (let c of candies) {
    if (c + extraCandies >= max) {
      result.push(true)
      continue
    }

    result.push(false)
  }

  return result
}

console.log(kidsWithCandies([2, 3, 5, 1, 3], 3));